config/config.go

```diff
@@ -65,34 +69,86 @@ func ReadConfig(configFilePath string) (*Configuration, *common.Block, error) {
 			return nil, nil, errors.Wrapf(err, "failed to read shared config, path is empty")
 		}
 	case "block":
-		if conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File != "" {
-			blockPath := conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File
-			data, err := os.ReadFile(conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File)
+		logger.Infof("reading shared config from block for %s node", nodeRole)
+		// If node is router or batcher, check if config store has blocks, and if yes bootstrap from the last block
+		if nodeRole == Router || nodeRole == Batcher {
+			configStore, err = configstore.NewStore(conf.LocalConfig.NodeLocalConfig.FileStore.Path)
 			if err != nil {
-				return nil, nil, fmt.Errorf("could not read block %s", blockPath)
+				return nil, nil, fmt.Errorf("failed creating %s config store: %s", nodeRole, err)
 			}
-			genesisBlock, err = protoutil.UnmarshalBlock(data)
+
+			listBlockNumbers, err := configStore.ListBlockNumbers()
 			if err != nil {
-				return nil, nil, fmt.Errorf("error unmarshalling to block: %s", err)
+				return nil, nil, fmt.Errorf("failed to list blocks from %s config store: %s", nodeRole, err)
 			}
 
-			consensusMetaData, err := ReadSharedConfigFromBootstrapConfigBlock(genesisBlock)
-			if err != nil {
-				return nil, nil, err
+			if len(listBlockNumbers) > 0 {
+				lastBlockNumber := uint64(len(listBlockNumbers) - 1)
```

Comment:
This is not the last block number, this is the index of the last...
The last block number is `realLastBlockNumber := listBlockNumbers[lastBlockNumber]`